### PR TITLE
Feature/prometheus exporter error log missing

### PR DESCRIPTION
**Pull request by ZhouZhongzhi**
Prometheus fails to generate logs when prometheus exporter produced a check exception occurs.

Zap.Logger is wrapped by wrapper mode to implement promhttp.Logger.

### CODE REVIEW

**Comment by ZhouZhongzhi** [comment]
The relevant MergeRequest
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6438

**Review by codeboten — COMMENTED** [review]
Pinging @Aneurysm9 as the codeowner

**Review by Aneurysm9 — APPROVED** [review]

File: exporter/prometheusexporter/log.go
Comment (jpkrohling) [inline exporter/prometheusexporter/log.go:12]: Does it have to be exported?

File: exporter/prometheusexporter/log.go
Comment (jpkrohling) [inline exporter/prometheusexporter/log.go:1]: The linter will probably complain, but you should add the license headers.

File: exporter/prometheusexporter/log.go
Comment (ZhouZhongzhi) [reply exporter/prometheusexporter/log.go:12]: If you don't export,Some exceptions may occur that cause indicators to be lost,But you don't have any way to check it

Comment (jpkrohling) [reply exporter/prometheusexporter/log.go:26]: Not sure I get what you mean... The only place using this new function is prometheus.go, which is part of the same package. You could certainly have it unexported (newLog) instead, from what I can see. What am I missing?

**Comment by ZhouZhongzhi** [comment]
Pinging @Aneurysm9  
My submission has been on hold for a long time, how do I merge this submission into main branch

File: exporter/prometheusexporter/log.go
Comment (Aneurysm9) [inline exporter/prometheusexporter/log.go:27]:
```suggestion
func newPromLogger(zapLog *zap.Logger) *promLogger {
```

Please make this unexported to avoid increasing the supported API surface area unnecessarily.

File: exporter/prometheusexporter/prometheus.go
Comment (Aneurysm9) [inline exporter/prometheusexporter/prometheus.go:60]:
```suggestion
				ErrorLog:      newPromLogger(set.Logger),
```

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]
Please make the requested changes to the exported API surface area.  Once the changes are made this can be tagged as ready to merge.

**Pull request closed without merging**
